lib/dart/lib/src/transport/t_message_reader.dart

```diff
@@ -0,0 +1,70 @@
+/// Licensed to the Apache Software Foundation (ASF) under one
+/// or more contributor license agreements. See the NOTICE file
+/// distributed with this work for additional information
+/// regarding copyright ownership. The ASF licenses this file
+/// to you under the Apache License, Version 2.0 (the
+/// "License"); you may not use this file except in compliance
+/// with the License. You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing,
+/// software distributed under the License is distributed on an
+/// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+/// KIND, either express or implied. See the License for the
+/// specific language governing permissions and limitations
+/// under the License.
+
+part of thrift;
+
+/// [TMessageReader] extracts a [TMessage] from bytes.  This is used to allow a
+/// transport to inspect the message seqid and map responses to requests.
+class TMessageReader extends TTransport {
```

Comment:
I get that this makes it easy to use a protocol to decode a message since protocols use `TTransport`s, but it does seem wrong to make `TMessageReader` extend `TTransport` given that it doesn't actually send or receive any data and it only supports reading. As of right now, I can't think of an alternative that wouldn't require going down a rabbit hole of refactoring, so I'm alright with this, but something to keep in mind.